src/search-and-replace-dialog.jsx

```diff
@@ -0,0 +1,286 @@
+/** @jsx React.DOM */
+
+var React = require('react');
+var Util = require("./util.js");
+
+var SearchAndReplaceDialog = React.createClass({
+
+    getDefaultProps: function() {
+        return {
+            onChange() {}    // TODO: throw an error
+        };
+    },
+
+    getInitialState() {
+        return {
+            replaceString: "",
+            searchResultCount: 0
+        }
+    },
+
+    updateSearchString(event) {
+        var searchIndex = 0;
+        var searchString = event.target.value;
+        var searchResultCount = this.getSearchResultCount(searchString);
+
+        this.setState({ searchResultCount });
+        this.props.onChange({ searchString, searchIndex });
+    },
+
+    updateReplaceString(event) {
+        this.setState({ replaceString: event.target.value });
+    },
+
+    updateSearchResults() {
+        var searchIndex = 0;
+        var searchResultCount = this.getSearchResultCount(this.props.searchString);
+
+        this.setState({ searchResultCount });
+        this.props.onChange({ searchIndex });
+    },
+
+    getSearchResultCount(searchString) {
```

Comment:
Good point.  Moving SearchAndReplaceDialog into each of the editors will make this easy to do.